### PR TITLE
DEV: Remove OpenComposer mixin and refactor related logic

### PR DESCRIPTION
**Pull request by davidtaylorhq**
The OpenComposer mixin comes from a time before we had a composer service. As well as being a general cleanup/refactor, this commit aims to removes interlinking between composer APIs and the discovery-related controllers which are being removed as part of #22622.

In summary, this commit:
- Removes OpenComposer mixin
- Adds and updates composer service APIs to support everything that `openComposer` did
- Updates consumers to call the composer service directly, instead of relying on the mixin (either directly, or via a route-action which bubbled up to some parent)
- Deprecates composer-related methods on `DiscourseRoute` and on the application route

### CODE REVIEW

File: app/assets/javascripts/discourse/app/services/composer.js [hunk 1]
```diff
@@ -567,33 +572,6 @@ export default class ComposerService extends Service {
     this.close();
   }
 
-  @action
-  async openComposer(options, post, topic) {
```
Comment (davidtaylorhq) [inline app/assets/javascripts/discourse/app/services/composer.js:571]: This function is very strangely named, especially given that we also have an `open` function. The only difference is that this one will prepend "Continued discussion from ..." to the composer.

To reduce confusion, I've added a 'prependText' option to the standard `open` function, and moved the "Continued discussion from..." text generation to the `<ComposerActions` component (the only place that was actually calling this function)

File: app/assets/javascripts/discourse/tests/fixtures/session-fixtures.js [hunk 2]
```diff
@@ -14,6 +14,7 @@ export default {
       moderator: true,
       staff: true,
       can_create_group: true,
+      can_create_topic: true,
```
Comment (davidtaylorhq) [inline app/assets/javascripts/discourse/tests/fixtures/session-fixtures.js:17]: This boolean has been present in the CurrentUserSerializer for a long time, but it was missing from this fixture

File: spec/system/composer/default_to_subcategory_spec.rb [hunk 3]
```diff
@@ -41,13 +41,9 @@
             end
           end
           describe "Category does not have subcategory" do
-            it "should have the 'New Topic' button enabled and default Subcategory set to latest default subcategory" do
+            it "should have the 'New Topic' button disabled" do
```
Comment (davidtaylorhq) [inline spec/system/composer/default_to_subcategory_spec.rb:44]: The two updated specs in this file seemed wrong to me. The `default_subcategory_on_read_only_category` should not affect the new-topic button behaviour on categories which have no subcategories.

File: app/assets/javascripts/discourse/app/models/category.js [hunk 4]
```diff
@@ -345,6 +345,16 @@ const Category = RestModel.extend({
   isUncategorizedCategory(id) {
     return Category.isUncategorized(id);
   },
+
+  get canCreateTopic() {
+    return this.permission === PermissionType.FULL;
+  },
+
+  get canCreateTopicSubcategory() {
+    return this.subcategories?.find(
```
Comment (CvX) [inline app/assets/javascripts/discourse/app/models/category.js:354]: strict typing 😁 

```suggestion
    return !!this.subcategories?.find(
```

Comment (CvX) [reply app/assets/javascripts/discourse/app/models/category.js:354]: Ah, it's a "can-create-topic subcategory" 😅 yeah, your suggestion sounds good!

**Pull request merged**